DataFormats/Common/interface/RefVector.h

```diff
@@ -77,7 +77,7 @@ namespace edm {
       key_type const& key = refVector_.keys()[idx];
       void const* memberPointer = refVector_.cachedMemberPointer(idx);
       if(memberPointer) {
-        const RefCore& newCore(core);
+        RefCore newCore(core);
```

Comment:
hi @Dr15Jones - please add // NOLINT to each line so that clang-tidy doesn't revert these back at some point (here, its not really running as you just changed a header file)

Reply:
Hi @Dr15Jones @davidlange6 

Late comment: I think it would be clearer to make a non-const version of `RefCore::setProductPtr()` (with a different name) for use here.  That would let the compiler and any other tools know that the copy can't be optimized away, communicating intent in a way that `// NOLINT` doesn't.